File: src/app/models/ExchangeRate.js

// Define collection and schema for ExchangeRate
export interface ExchangeRate {
    baseCurrency:
	type : string
    counterCurrency:
	type : string
    rate:
	type : String
    asOf:
	type : Date
    source:
	type : string
    Bank:
	type : Schema.Types.ObjectId
    FxTrades:
 	type : [{ type: Schema.Types.ObjectId, ref: 'FXTrade' }]
#
    collection: 'exchangeRates'
}

File: bankingonaspdotnet/Domain/ExchangeRate.cs

using bankingonaspdotnet.Contracts;

namespace bankingonaspdotnet.Domain;

public class ExchangeRate
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public virtual long? ExchangerateId { get; set; }
    public virtual string? BaseCurrency { get; set; }
    public virtual string? CounterCurrency { get; set; }
    public virtual decimal? Rate { get; set; }
    public virtual DateOnly? AsOf { get; set; }
    public virtual string? Source { get; set; }
    public virtual Bank? Bank { get; set; }
    public virtual ICollection<FXTrade> FxTrades { get; set; } = new List<FXTrade>();

    public static ExchangeRate FromRequest(ExchangeRateRequest request)
    {
        return new ExchangeRate
        {
            Id = request.Id,
            BaseCurrency = request.BaseCurrency,
            CounterCurrency = request.CounterCurrency,
            Rate = request.Rate,
            AsOf = request.AsOf,
            Source = request.Source,
        };
    }
}
